vint/ast/parsing.py

```diff
@@ -42,23 +34,15 @@ def parse(self, string):
 
     def parse_file(self, file_path):
         """ Parse vim script file and return the AST. """
-        with file_path.open(mode='rb') as f:
-            bytes_seq = f.read()
-
-            is_empty = len(bytes_seq) == 0
-            if is_empty:
-                return self.parse('')
-
-            encoding_hint = chardet.detect(bytes_seq)
-            encoding = encoding_hint['encoding']
-            if not encoding:
-                # Falsey means we cannot detect the encoding of the file.
-                raise EncodingDetectionError(file_path)
-
-            decoded = bytes_seq.decode(encoding)
-            decoded_and_lf_normalized = decoded.replace('\r\n', '\n')
+        decoder = Decoder(default_decoding_strategy)
+        decoded = decoder.read(file_path)
+        decoded_and_lf_normalized = decoded.replace('\r\n', '\n')
 
+        try:
             return self.parse(decoded_and_lf_normalized)
+        except vimlparser.VimLParserException:
+            pprint(decoder.debug_hint)
```

Comment:
I think this should only get done in verbose/debug mode?

Reply:
Oops, it is debug code as you said.
I'm gonna remove it at next release.